homeassistant/components/jewish_calendar/sensor.py

```diff
@@ -160,29 +166,29 @@ def __init__(self, data, description: SensorEntityDescription) -> None:
         self._candle_lighting_offset = data["candle_lighting_offset"]
         self._havdalah_offset = data["havdalah_offset"]
         self._diaspora = data["diaspora"]
-        self._state = None
+        self._state: datetime | None = None
         self._holiday_attrs: dict[str, str] = {}
 
     @property
-    def native_value(self):
+    def native_value(self) -> StateType:
         """Return the state of the sensor."""
         if isinstance(self._state, datetime):
             return self._state.isoformat()
         return self._state
 
-    async def async_update(self):
+    async def async_update(self) -> None:
         """Update the state of the sensor."""
         now = dt_util.now()
         _LOGGER.debug("Now: %s Location: %r", now, self._location)
 
         today = now.date()
         sunset = dt_util.as_local(
-            get_astral_event_date(self.hass, SUN_EVENT_SUNSET, today)
+            get_astral_event_date(self.hass, SUN_EVENT_SUNSET, today)  # type: ignore[arg-type]
```

Comment:
The comment was referring to this line:
```python
get_astral_event_date(self.hass, SUN_EVENT_SUNSET, today)  # type: ignore[arg-type]`
```

if you type the `today` as @janiversen suggested you don't need the `# type: ignore[arg-type]`
`today: dt.datetime = ...`

Reply:
I sure tried, however the type is Optional so I recieve an error
`homeassistant/components/jewish_calendar/sensor.py:186: error: Argument 1 to "as_local" has incompatible type "Optional[datetime]"; expected "datetime"`

Reply:
So looking at the code of [get_astral_event_date](https://github.com/home-assistant/core/blob/2f960e558f025d0e5f414c4e54058f96ff4e33dd/homeassistant/helpers/sun.py#L104) it return `datetime.datetime | None`, so the problem was not related to the `today` untyped, but it is a real potential bug since if `get_astral_event_date` will return `None` and you pass it to `as_local` it will raise an exception.

A clean solution would be to stop processing and maybe log an error, something like this:
```python
if event_date := get_astral_event_date(self.hass, SUN_EVENT_SUNSET, today) is None:
   _LOGGER.error("Can't get sunset event date for %s", today)
   return

sunset = dt_util.as_local(event_date)
```

Reply:
Thanks a lot